### PR TITLE
YLW 3119:  2 Add library methods

### PR DESCRIPTION
**Pull request by davidgreenland**
#### What 
Extended the Library class to expose a List method that returns catalogue.
Added EnchantedLibraryApp class.

#### Why
To add the required functionality to the Library while adhering to SOLID principles.

#### How
- Created a ILibraryUserInteraction interface and a LibraryConsoleUserInteraction class to implement it. 
- Injected this along with the Library into the App.
- Created a PrintBooks method in the ConsoleUserInteraction class that formats a table of books and prints it to the console.

### CODE REVIEW

File: TheEnchantedLibrary/Models/Library.cs [hunk 1]
```diff
@@ -12,4 +12,10 @@ public Library(List<Book> books)
     {
         Catalogue = books;
     }
-}
\ No newline at end of file
+
+    public ICollection<Book> List()
+    {
+        Console.WriteLine("Printing books");
```
Comment (SiddiqueAbdullah) [inline TheEnchantedLibrary/Models/Library.cs]: remove

File: TheEnchantedLibrary/Models/Library.cs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ﻿namespace TheEnchantedLibrary.Models;
 
-public class Library
+public class Library : ILibrary
 {
     private ICollection<Book> Catalogue = [];
```
Comment (SiddiqueAbdullah) [inline TheEnchantedLibrary/Models/Library.cs]: _books

Comment (davidgreenland) [reply TheEnchantedLibrary/Models/Library.cs]: done

File: TheEnchantedLibrary/Models/Library.cs (hunk 1 above)
Comment (davidgreenland) [reply TheEnchantedLibrary/Models/Library.cs]: removed

**Review by SiddiqueAbdullah — APPROVED** [review]

**Pull request merged**